app/src/services/cpk.ts

```diff
@@ -17,6 +17,13 @@ interface CPKBuyOutcomesParams {
 }
 
 class CPKService {
+  static getCPK = async (provider: any) => {
```

Comment:
Wouldn't it better to make a constructor, and then make `buyOutcomes` an instance method?

Reply:
> Wouldn't it better to make a constructor, and then make `buyOutcomes` an instance method?

@fvictorio Mmmm I think we can't implement an async constructor(is not allowed, please correct me if I'm wrong), maybe and initialize async method ?

Reply:
Yep, you are right. A static async factory method is probably the best option:

```
constructor(cpk) {
  this.cpk = cpk
}

static async create() {
  const cpk = await ...
  return new CPKService(cpk)
}

buyOutcomes = async (...) { // instance method that uses this.cpk
```

Reply:
Done!